sycl/test/format.py

```diff
@@ -0,0 +1,107 @@
+import lit.Test
+import lit.util
+import lit.formats
+
+import os
+import re
+
+
+class SYCLHeadersTest(lit.formats.TestFormat):
+    def getTestsForPath(
+        self, testSuite, path_in_suite, filepath, litConfig, localConfig
+    ):
+        yield lit.Test.Test(testSuite, path_in_suite, localConfig, file_path=filepath)
+
+    def getTestsInDirectory(self, testSuite, path_in_suite, litConfig, localConfig):
+        # We traverse build/sycl/include/sycl directory
+        source_path = os.path.join(localConfig.sycl_include, "sycl")
+
+        # Optional filter can be passed through command line options
+        headers_filter = localConfig.sycl_headers_filter
+        for dirpath, _, filenames in os.walk(source_path):
+            relative_dirpath = dirpath[len(localConfig.sycl_include) + 1 :]
+            for filename in filenames:
+                if not filename.endswith(".hpp"):
+                    continue
+                filepath = os.path.join(dirpath, filename)
+
+                if headers_filter is not None:
+                    # Skip headers that doesn't match passed regexp
+                    if re.match(headers_filter, filepath) is None:
+                        continue
+                for t in self.getTestsForPath(
+                    testSuite,
+                    path_in_suite
+                    + (
+                        relative_dirpath,
+                        filename,
+                    ),
+                    filepath,
+                    litConfig,
+                    localConfig,
+                ):
+                    yield t
+
+    def execute(self, test, litConfig):
+        # List of files that are not expected to compile cleanly when used
+        # standalone. `os.path.join` is required here so the filtering works
+        # cross-platform
+        xfail = [
+            os.path.join(
+                "sycl", "ext", "oneapi", "experimental", "backend", "cuda.hpp"
+            ),
+            os.path.join(
+                "sycl", "ext", "intel", "esimd", "detail", "types_elementary.hpp"
+            ),
+        ]
+
+        command = [
+            test.config.clang,
+            "-fsycl",
+            "-fsyntax-only",
+            "-include",
+            test.file_path,
+            os.path.join(
+                test.suite.getSourcePath(("self-contained-headers",)),
+                "Inputs",
+                "test.cpp",
+            ),
+        ]
+
+        is_xfail = False
+        for path in xfail:
+            if test.file_path.endswith(path):
+                is_xfail = True
+                break
+
+        try:
+            out, err, exitCode = lit.util.executeCommand(
+                command,
+                # cwd=???,
```

Comment:
delete this or pass in an explicit directory

Reply:
Replaced with a `TODO` in cf18fe0e60ab1a34251c8f464aaced57fef4bd38, because I'm not entirely sure what is correct value for the argument